backend/tests/integration/controllers/posts-controller.test.js

```diff
@@ -0,0 +1,161 @@
+    expect(fetchedPost).not.toBeNull();
+    expect(fetchedPost.title).toBe(createPostObject().title);
+  });
+
+  it('Post creation: Failure - Missing required fields', async () => {
+    const postObject = createPostObject();
+    delete postObject.title;
+    const response = await request(server).post('/api/posts').send(postObject);
+
+    expect(JSON.parse(response.text)).toEqual({ message: 'All fields are required.' });
+    expect(response.status).toBe(400);
+  });
+
+  it('Post creation: Failure - Invalid category count', async () => {
+    const postObject = createPostObject({
+      categories: [validCategories[0], validCategories[1], validCategories[2], validCategories[3]],
+    });
+    const response = await request(server).post('/api/posts').send(postObject);
+
+    expect(JSON.parse(response.text)).toEqual({
+      message: 'Please select up to three categories only.',
+    });
+    expect(response.status).toBe(400);
+  });
+
+  it('Post creation: Failure - Invalid image URL format', async () => {
+    const postObject = createPostObject({
+      imageLink: 'https://www.invalid-image.gif',
+    });
+    const response = await request(server).post('/api/posts').send(postObject);
+
+    expect(JSON.parse(response.text)).toEqual({
+      message: 'Image URL must end with .jpg, .jpeg, or .png',
+    });
+    expect(response.status).toBe(400);
+  });
+
+  it('Post creation: Failure - Internal server error', async () => {
+    // Mocking a scenario where the server encounters an internal error during post creation
+    jest.spyOn(Post.prototype, 'save').mockRejectedValueOnce(new Error('Internal Server Error'));
+
+    const postObject = createPostObject();
+    const response = await request(server).post('/api/posts').send(postObject);
+
+    expect(response.status).toBe(500);
+    expect(JSON.parse(response.text)).toEqual({ message: 'Internal Server Error' });
+  });
+});
+describe('Integration Tests: Get all posts', () => {
+  it('Get all posts: Success', async () => {
+    const response = await request(server).get('/api/posts');
+
+    expect(response.status).toBe(200);
+    expect(response.body).toBeInstanceOf(Array);
+  });
+});
+describe('Integration Tests: Get all posts by category', () => {
+  it('Get all posts by category: Success', async () => {
+    const category = validCategories[0];
+    const response = await request(server).get(`/api/posts/categories/${category}`);
+
+    expect(response.status).toBe(200);
+    expect(response.body).toBeInstanceOf(Array);
+  });
+
+  it('Get all posts by category: Failure - Invalid category', async () => {
+    const category = 'invalid-category';
+    const response = await request(server).get(`/api/posts/categories/${category}`);
+
+    expect(response.status).toBe(400);
+    expect(JSON.parse(response.text)).toEqual({
+      message: 'Invalid category',
+    });
+  });
+});
+describe('Integration Tests: Get all featured posts', () => {
```

Comment:
Both test blocks appear to perform same operations, which leads to redundancy.
Can you consider to merging these tests into a single one or make test separate for retrieving featured and latest posts only?

Reply:
Hey @hemant14050, thank you for the catch, I missed it, fixed it :)